File: 3rd Semester (C#)/Lab4/Banks/Models/PassportSeries.cs
using Banks.Tools;

namespace Banks.Models;

public class PassportSeries
{
    private const int MinSeriesNumber = 1000;
    private const int MaxSeriesNumber = 9999;

    public PassportSeries(int series)
    {
        if (series < MinSeriesNumber || series > MaxSeriesNumber)
        {
            throw new BanksException($"Given value series: {series} has to be between {MinSeriesNumber} and {MaxSeriesNumber}");
        }

        Series = series;
    }

    public int Series { get; }
}

File: 4th Semester (Java)/lab-1/src/main/java/models/PassportSeries.java
package models;

import exceptions.BanksException;

/**
 * Contains information about series of passport.
 */
public class PassportSeries {
    private final int MinSeriesNumber = 1000;
    private final int MaxSeriesNumber = 9999;

    /**
     * Passport series.
     */
    private final int series;

    /**
     * PassportNumber constructor.
     * @param series passport series
     * @throws BanksException if given passport number is not in range of [1000; 9999]
     */
    public PassportSeries(int series) throws BanksException {
        if (series < MinSeriesNumber || series > MaxSeriesNumber) {
            throw new BanksException("Given value series has to be between " + MinSeriesNumber + " and " + MaxSeriesNumber);
        }

        this.series = series;
    }

    public int getSeries() {
        return series;
    }
}
